modules/git/log_name_status.go

```diff
@@ -56,6 +56,10 @@ func LogNameStatusRepo(ctx context.Context, repository, head, treepath string, p
 	} else if treepath != "" {
 		files = append(files, treepath)
 	}
+	// Use the :(literal) pathspec magic to handle edge cases with files named like ":file.txt" or "*.jpg"
+	for i, file := range files {
+		files[i] = ":(literal)" + file
+	}
 	cmd.AddDashesAndList(files...)
```

Comment:
Hmm, wouldn't it be better to add it to `AddDashesAndList` directly?
That way, it will already be applied to every possible instance, I don't know of any usage that expects wildcards to work, and it is in the other cases **not** intended to be used as a wildcard.

Reply:
> Hmm, wouldn't it be better to add it to `AddDashesAndList` directly?

This approach has been discussed before, it might break something.

Maybe next time we can introduce `AddDashesAndLiterals`, but not change the behavior of `AddDashesAndList`.